src/Server.php

```diff
@@ -2,336 +2,17 @@
-
-        $that = $this;
-        $this->streamingServer->on('error', function ($error) use ($that) {
-            $that->emit('error', array($error));
-        });
-    }
-
-    /**
-     * Starts listening for HTTP requests on the given socket server instance
-     *
-     * The given [`React\Socket\ServerInterface`](https://github.com/reactphp/socket#serverinterface)
-     * is responsible for emitting the underlying streaming connections. This
-     * HTTP server needs to be attached to it in order to process any
-     * connections and pase incoming streaming data as incoming HTTP request
-     * messages. In its most common form, you can attach this to a
-     * [`React\Socket\Server`](https://github.com/reactphp/socket#server) in
-     * order to start a plaintext HTTP server like this:
-     *
-     * ```php
-     * $server = new React\Http\Server($handler);
-     *
-     * $socket = new React\Socket\Server(8080);
-     * $server->listen($socket);
-     * ```
-     *
-     * See also [hello world server example](../examples/51-server-hello-world.php)
-     * for more details.
-     *
-     * This example will start listening for HTTP requests on the alternative
-     * HTTP port `8080` on all interfaces (publicly). As an alternative, it is
-     * very common to use a reverse proxy and let this HTTP server listen on the
-     * localhost (loopback) interface only by using the listen address
-     * `127.0.0.1:8080` instead. This way, you host your application(s) on the
-     * default HTTP port `80` and only route specific requests to this HTTP
-     * server.
-     *
-     * Likewise, it's usually recommended to use a reverse proxy setup to accept
-     * secure HTTPS requests on default HTTPS port `443` (TLS termination) and
-     * only route plaintext requests to this HTTP server. As an alternative, you
-     * can also accept secure HTTPS requests with this HTTP server by attaching
-     * this to a [`React\Socket\Server`](https://github.com/reactphp/socket#server)
-     * using a secure TLS listen address, a certificate file and optional
-     * `passphrase` like this:
-     *
-     * ```php
-     * $server = new React\Http\Server($handler);
-     *
-     * $socket = new React\Socket\Server('tls://0.0.0.0:8443', null, array(
-     *     'local_cert' => __DIR__ . '/localhost.pem'
-     * ));
-     * $server->listen($socket);
-     * ```
-     *
-     * See also [hello world HTTPS example](../examples/61-server-hello-world-https.php)
-     * for more details.
-     *
-     * @param ServerInterface $socket
-     */
-    public function listen(ServerInterface $socket)
-    {
-        $this->streamingServer->listen($socket);
-    }
-
-    /**
-     * @param string $memory_limit
-     * @param string $post_max_size
-     * @return ?int
-     */
-    private function getConcurrentRequestsLimit($memory_limit, $post_max_size)
+    class Server extends HttpServer
```

Comment:
```suggestion
    final class Server extends HttpServer
```

Reply:
Good catch, updated as suggested :+1:

The class definition isn't actually executed because it's essentially in an `if(false)` block, but still makes sense for static analysis and IDE autocompletion :+1: